### PR TITLE
Add rule and test for GuardDuty

### LINKED ISSUE #503: Feature: Support GuardDuty
Amazon just announced a new security service named GuardDuty this week. https://aws.amazon.com/guardduty/

It has some useful rules that would be difficult to implement in StreamAlert, such as detecting logins from new ASNs, which requires correlation with past events.  When GuardDuty detects something, it writes to CloudWatch Events.  My request is:
1. Support the schema used by GuardDuty alerts.
2. Create a StreamAlert rule to alert on any GuardDuty alerts.

From the User Guide (p 44, https://docs.aws.amazon.com/guardduty/latest/ug/guardduty-ug.pdf), the format of the CloudWatch events is:
```
{
 "version": "0",
 "id": "cd2d702e-ab31-411b-9344-793ce56b1bc7",
 "detail-type": "GuardDuty Finding",
 "source": "aws.guardduty",
 "account": "111122223333",
 "time": "1970-01-01T00:00:00Z",
 "region": "us-east-1",
 "resources": [],
 "detail": {COMPLETE_GUARDDUTY_FINDING_JSON}
 } 
```

### PR DESCRIPTION
**Pull request by 0xdabbad00**
to: @airbnb/streamalert-maintainers
size: small
resolves #503 

## Background

Adds a rule for GuardDuty and a test for it

## Testing

Setup StreamAlert. Enabled GuardDuty. Added this.

### CODE REVIEW

File: rules/community/guardduty/guardduty.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    author:         spiper
+    description:    Alert on GuardDuty events
+    playbook:       (a) identify the AWS account in the log
+                    (b) identify what resource(s) are impacted
+                    (c) contact the point-of-contact for the account
+    testing:        From the GuardDuty AWS page (https://console.aws.amazon.com/guardduty/home)
+                    click the button to "Generate Sample Findings"
+    """
+
+    if rec['detail-type'] == 'GuardDuty Finding':
```
Comment (jacknagz) [inline rules/community/guardduty/guardduty.py]: Hey @0xdabbad00, I would consider moving this logic into a `matcher`, and then the GuardDuty checks can be applied to multiple rules.  That would also simplify this example rule

**Comment by jacknagz** [comment]
Thanks for the contribution @0xdabbad00! Looks like there was a CI failure here:

```
The command "sphinx-build -W docs/source docs/build" exited with 0.
59.78s$ ./tests/scripts/pylint.sh
Starting pylint script
************* Module rules.community.guardduty.guardduty
W:  2, 0: Unused in_set imported from helpers.base (unused-import)
------------------------------------
Your code has been rated at 10.00/10
```

File: matchers/matchers.py [hunk 2]
```diff
@@ -16,3 +16,7 @@
 from stream_alert.rule_processor.rules_engine import StreamRules
 
 matcher = StreamRules.matcher()
+
+@matcher
+def guardduty(record):
```
Comment (jacknagz) [inline matchers/matchers.py]: nit: `guard_duty` instead of `guardduty`

File: rules/community/guardduty/guardduty.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+from stream_alert.rule_processor.rules_engine import StreamRules
+
+rule = StreamRules.rule
+disable = StreamRules.disable()
+
+
+@rule(logs=['cloudwatch:events'],
+      matchers=['guardduty'],
+      outputs=['slack:sample-channel'])
+def guardduty(*_):
```
Comment (jacknagz) [inline rules/community/guardduty/guardduty.py]: let's call this rule `guard_duty_all`

**Comment by 0xdabbad00** [comment]
I tried to do some renaming of `guardduty` to `guard_duty`, but I don't know where all to apply that renaming, as we have CloudTrail as `cloudtrail` and StreamAlert as both `streamalert` and `stream_alert`.  "GuardDuty" is the name of the service (one word, camel cased).

**Comment by 0xdabbad00** [comment]
For example, I now have a file in the path `rules/community/guardduty/guard_duty_all.py` which looks weird to have renamed one (the file), but not the other (the directory), but within the `rules/community` directory, a file listing shows:
```
cloudtrail/
guardduty/
```
So it would look weird to rename that directory.

**Comment by jacknagz** [comment]
FYI @airbnb/streamalert-maintainers

**Pull request merged**